src/flyte/_tools.py

```diff
@@ -22,7 +21,10 @@ def is_in_cluster() -> bool:
     Check if the task is running in a cluster
     :return is_in_cluster (bool): True or False
     """
-    if os.getenv("_UN_CLS"):
+    from flyte._context import internal_ctx
+
+    ctx = internal_ctx()
+    if ctx.data.task_context and ctx.data.task_context.mode == "remote":
```

Comment:
interesting, we should make sure this is set correctly all the time.

Reply:
just use flyte.ctx() and you can add `is_remote` or `is_in_cluster` to TaskContext. No more exposing weird _ vars